### PR TITLE
Add utility function source_files()

### PR DESCRIPTION
**Pull request by Smit-tay**
Add associated testthat file

### CODE REVIEW

**Comment by Smit-tay** [comment]
This is a slightly  modified version of what I previously sent.

File: R/utility.R [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  files <- files[file.info(files)$isdir == FALSE]
+
+  # Filter out any files which match the File exclude regex
+  if (!is.null(exclude_files)) {
+    files <- files[!grepl(exclude_files, basename(files))]
+  }
+  
+  # Filter out any directories which match the Directory exclude regex
+  if (!is.null(exclude_dirs)) {
+    files <- files[!grepl(exclude_dirs, dirname(files))]
```
Comment (RobertASmith) [inline R/utility.R:44]: I think this should be a grep on the relative path (i.e. removing everything before the directory)

e.g.
```
  # Filter out any directories which might match the Directory exclude regex
  if (!is.null(dirExclRegx)) {
    # the directory exclusion is from the path provided, not the full path
    file_path_within_directory <- gsub(pattern = path, replacement =  "", x = dirname(files))
    files <- files[!grepl(dirExclRegx, file_path_within_directory) ]
  }
```

Comment (RobertASmith) [reply R/utility.R:44]: Although this is not a great solution, should ideally not use `gsub` and instead remove the start of the string that equals the path string.

File: R/utility.R [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+#' \dontrun{
+#' source_files(file_regx = ".*",  ## any file name
+#'  dir_regx = ".*",   # the current directory and all subdirectories
+#'  recursive = FALSE  # don't recurse
+#'  exclude_files = ".*utility.*",   # don't source any file with the string "utility" anywhere in its basename
+#'  exclude_dirs = "\/tmp\/"  # don't source any files which reside in any directory named "tmp", or its subdirectories
+#' )
+#' }
+#'
+source_files <- function(file_regx = ".*",  dir_regx = ".", recursive = TRUE, exclude_files = NULL, exclude_dirs = NULL, local=!testthat::is_testing() ) {
```
Comment (RobertASmith) [inline R/utility.R:28]: would be good to have verbose option - see #43

[the remark this answers is not in the document]
Comment (Smit-tay) [reply R/utility.R:28]: true enough.  In fact, it's not a regex parameter to list.files(), it's a character vector of full path names

File: R/utility.R [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+#' \dontrun{
+#' source_files(file_regx = ".*",  ## any file name
+#'  path = ".*",   # the current directory and all subdirectories
+#'  recursive = FALSE  # don't recurse
+#'  exclude_files = ".*utility.*",   # don't source any file with the string "utility" anywhere in its basename
+#'  exclude_dirs = "\/tmp\/"  # don't source any files which reside in any directory named "tmp", or its subdirectories
+#' )
+#' }
+#'
+source_files <- function( file_regx = ".*",
```
Comment (RobertASmith) [inline R/utility.R:30]: By default just want to source .R files only

**Comment by RobertASmith** [comment]
See diff [here](https://github.com/dark-peak-analytics/assertHE/pull/43/files) for the other pull request for some extra bits needed:

1) remove miceadds dependency from description file
2) can now use the new function (source_files) in `R/project_visualiser.R` rather than the miceadds version, and remove the importFrom as well.

I think this then completely removes the miceadds dependency.

**Comment by Smit-tay** [comment]
Status ?

**Comment by RobertASmith** [comment]
All items addressed, merged into main.

**Pull request merged**